### PR TITLE
Optimizer refactoring

### PR DESCRIPTION
**Pull request by ftdebugger**

### CODE REVIEW

**Comment by ftdebugger** [comment]
@jakwuh @next0 what you think?

File: lib/build.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+export let plugins = {
+    TemplateRequirePlugin,
+    TemplateCSSModulesPlugin,
+    TemplateFlattenPlugin,
+    TemplateTrimSpacesPlugin
+};
+
+export let ComponentsScanner = Scanner;
+export let TemplateCompiler = Compiler;
```
Comment (jakwuh) [inline lib/build.js]:
```js
export {TemplateCompiler} from './Template/TemplateCompiler';
export {ComponentsScanner} from './Build/ComponentsScanner';
```

**Comment by next0** [comment]
👍

**Comment by next0** [comment]
May be we should rename `TemplateOptimizer` to something like `TemplateTransformer` because we can write plugging not only for optimize code (we can add new functionality). In that case method `optimize` we should rename too.

**Comment by ftdebugger** [comment]
Fixed

**Pull request merged**